lib/facter/self_service.rb

```diff
@@ -241,3 +254,30 @@
     { S0040: PuppetSelfService.service_running_enabled('puppet_system_processes-metrics.timer') }
   end
 end
+
+# Facter.add('maximum') do
+#  setcode do
+#    PuppetSelfService.max_connections.map(&:to_i)
+#  end
+# end
+
+# Facter.add('current') do
+#  setcode do
+#    PuppetSelfService.cur_connections.map(&:to_i)
+#  end
+# end
+
+Facter.add('approaching_limit') do
+  require_relative '../shared/puppet_self_service'
+  approaching_limit = false
+  maximum = PuppetSelfService.max_connections.to_i
+  current = PuppetSelfService.cur_connections.to_i
+  percent_used = current / maximum * 100
+  if percent_used >= 90
+    approaching_limit = true
+  end
+
+  setcode do
+    approaching_limit
+  end
+end
```

Comment:
Is this separate fact left over from your development process? this would need removed in favour of the fact chunk in S0029